alexandria/core/views.py

```diff
@@ -280,3 +284,42 @@ def check_object_permissions(self, request, instance):
         # call DGAP
         self._check_object_permissions(request, instance)
         super().check_object_permissions(request, instance)
+
+
+class SearchViewSet(
+    PermissionViewMixin,
+    VisibilityViewMixin,
+    AutoPrefetchMixin,
+    PreloadIncludesMixin,
+    RelatedMixin,
+    ListModelMixin,
+    GenericViewSet,
+):
+    serializer_class = serializers.FileSerializer
+    queryset = models.File.objects.all()
+
+    def filter_queryset(self, queryset):
+        query = self.request.query_params.get("filter[query]")
+
+        search_queries = []
+        for lang, desc in settings.LANGUAGES:
+            config = settings.ISO_639_TO_PSQL_SEARCH_CONFIG.get(lang, "simple")
+            search_queries.append(
+                Q(language=lang)
+                & Q(
+                    content_vector=SearchQuery(
+                        query, config=config, search_type="phrase"
+                    )
+                )
+            )
+
+        search_query = reduce(or_, search_queries)
+
+        latest_files = models.File.objects.filter(
+            document=OuterRef("document_id"), variant=models.File.Variant.ORIGINAL
+        ).order_by("-created_at")
+        queryset = queryset.filter(pk=Subquery(latest_files.values("pk")[:1])).filter(
+            search_query
+        )
```

Comment:
Why do we need a subquery?

Reply:
Current the search only searches over latest original files. This subquery is for that.

Reply:
Aaah now I get it.. Nevermind.